org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/JavaFormatterTestHelper.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.formatting2.internal;
+
+import org.eclipse.xtext.formatting2.IFormatter2;
+import org.eclipse.xtext.testing.formatter.FormatterTestHelper;
+import org.eclipse.xtext.testing.formatter.FormatterTestRequest;
+
+import com.google.inject.Inject;
+
+/**
+ * @author Arne Deutsch - Initial contribution and API
+ */
+public class JavaFormatterTestHelper extends FormatterTestHelper {
+
+	@Inject
+	private FormatterTestLanguageJavaFormatter formatter;
```

Comment:
Can this become a reusable utility if we inject the interface of the formatter instead of the concrete type?

Reply:
Not sure. The paretn does the following:

```
@Inject(optional = true)
private Provider<IFormatter2> formatter;

protected IFormatter2 createFormatter(FormatterTestRequest request) {
	checkNotNull(formatter, "There is a Guice Binding missing for " + IFormatter2.class.getName());
	return formatter.get();
}
```

And the other sibling overrides with:

```
@Override
protected IFormatter2 createFormatter(FormatterTestRequest req) {
	return ((GenericFormatterTestRequest) req).getFormatter();
}
```

Right now it seems there is not much room for reusability here ...